src/databricks/labs/ucx/source_code/whitelist.py

```diff
@@ -22,16 +20,13 @@
 logger = logging.getLogger(__name__)
 
 
-class WhitelistResolver(BaseImportResolver):
+class WhitelistResolver:
 
-    def __init__(self, whitelist: Whitelist, next_resolver: BaseImportResolver | None = None):
-        super().__init__(next_resolver)
+    def __init__(self, whitelist: Whitelist):
+        super().__init__()
         self._whitelist = whitelist
 
-    def with_next_resolver(self, resolver: BaseImportResolver) -> BaseImportResolver:
-        return WhitelistResolver(self._whitelist, resolver)
-
-    def resolve_import(self, path_lookup: PathLookup, name: str) -> MaybeDependency:
+    def resolve_import(self, name: str) -> MaybeDependency | None:
         # TODO attach compatibility to dependency, see https://github.com/databrickslabs/ucx/issues/1382
         compatibility = self._whitelist.compatibility(name)
```

Comment:
move the body of this method as private method to `ImportResolver` class and get rid of `WhitelistResolver`.

Reply:
done